### PR TITLE
fix: log when outputting preempt error

### PR DESCRIPTION
**Pull request by googs1025**
#### What type of PR is this?
log when outputting preempt error

#### What this PR does / why we need it:
- log when outputting preempt error：preempt should be an important method. If an error occurs, a log should be printed.
- refector the judgment statement of allocate action to make it more readable
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: pkg/scheduler/actions/preempt/preempt.go [hunk 1]
```diff
@@ -254,19 +259,20 @@ func preempt(
 			klog.V(3).Infof("No validated victims on Node <%s>: %v", node.Name, err)
 			continue
 		}
-
-		victimsQueue := util.NewPriorityQueue(func(l, r interface{}) bool {
+		// lower priority task are evicted first
```
Comment (lowang-bh) [inline pkg/scheduler/actions/preempt/preempt.go:262]: Those code have been refacted in another [PR sort victims in reclaim action #3389](https://github.com/volcano-sh/volcano/pull/3389/files#diff-3ef0cda85fbd1a9e76f3e873cad14fbe0cbe440718d999d0ff188413912c558aL258-R258)

Comment (googs1025) [reply pkg/scheduler/actions/preempt/preempt.go:262]: That sounds good to me, I was just about to make changes to this part of the code. I'll revert this part back later.

File: pkg/scheduler/actions/allocate/allocate.go [hunk 2]
```diff
@@ -69,16 +69,15 @@ func (alloc *Action) pickUpQueuesAndJobs(queues *util.PriorityQueue, jobsMap map
 	ssn := alloc.session
 	for _, job := range ssn.Jobs {
 		// If not config enqueue action, change Pending pg into Inqueue statue to avoid blocking job scheduling.
-		if conf.EnabledActionMap["enqueue"] {
-			if job.IsPending() {
+		if job.IsPending() {
+			if conf.EnabledActionMap["enqueue"] {
 				klog.V(4).Infof("Job <%s/%s> Queue <%s> skip allocate, reason: job status is pending.",
```
Comment (Monokaix) [inline pkg/scheduler/actions/allocate/allocate.go:74]: Should also `continue` here.

Comment (googs1025) [reply pkg/scheduler/actions/allocate/allocate.go:74]: thanks for the reminding :), done

File: pkg/scheduler/actions/preempt/preempt.go [hunk 3]
```diff
@@ -130,8 +131,9 @@ func (pmpt *Action) Execute(ssn *framework.Session) {
 					}
 					// Preempt other jobs within queue
 					return job.Queue == preemptorJob.Queue && preemptor.Job != task.Job
-				}, ph); preempted {
-					assigned = true
+				}, ph)
+				if err != nil {
+					klog.V(3).Infof("Failed to preempt Task , err: %s", err)
```
Comment (lowang-bh) [inline pkg/scheduler/actions/preempt/preempt.go:136]: We'd better add current task name info in log

File: pkg/scheduler/actions/preempt/preempt.go [hunk 4]
```diff
@@ -179,6 +181,9 @@ func (pmpt *Action) Execute(ssn *framework.Session) {
 					// Preempt tasks within job.
 					return preemptor.Job == task.Job
 				}, ph)
+				if err != nil {
+					klog.V(4).Infof("Failed to preempt Task , err: %s", err)
```
Comment (lowang-bh) [inline pkg/scheduler/actions/preempt/preempt.go:185]: same as above.

File: pkg/scheduler/actions/preempt/preempt.go (hunk 3 above)
Comment (googs1025) [reply pkg/scheduler/actions/preempt/preempt.go]: make sense to me, done

File: pkg/scheduler/actions/preempt/preempt.go [hunk 5]
```diff
@@ -130,8 +131,9 @@ func (pmpt *Action) Execute(ssn *framework.Session) {
 					}
 					// Preempt other jobs within queue
 					return job.Queue == preemptorJob.Queue && preemptor.Job != task.Job
-				}, ph); preempted {
-					assigned = true
+				}, ph)
+				if err != nil {
+					klog.V(3).Infof("Failed to preempt Task <%s/%s> , err: %s", preemptor.Namespace, preemptor.Name, err)
```
Comment (lowang-bh) [inline pkg/scheduler/actions/preempt/preempt.go:136]: The log is ambiguous, the preemptor is a task to preempt others, not the one be preempted.

File: pkg/scheduler/actions/preempt/preempt.go [hunk 6]
```diff
@@ -179,6 +181,9 @@ func (pmpt *Action) Execute(ssn *framework.Session) {
 					// Preempt tasks within job.
 					return preemptor.Job == task.Job
 				}, ph)
+				if err != nil {
+					klog.V(3).Infof("Failed to preempt Task <%s/%s> , err: %s", preemptor.Namespace, preemptor.Name, err)
```
Comment (lowang-bh) [inline pkg/scheduler/actions/preempt/preempt.go:185]: same as above.

File: pkg/scheduler/actions/preempt/preempt.go (hunk 5 above)
Comment (googs1025) [reply pkg/scheduler/actions/preempt/preempt.go]: i change to `klog.V(3).Infof("Preemptor <%s/%s> failed to preempt Task , err: %s", preemptor.Namespace, preemptor.Name, err)
`

**Comment by googs1025** [comment]
@lowang-bh @Monokaix /PTAL thanks!

**Pull request merged**